magellan/blocker/attr_equiv_blocker.py

```diff
@@ -38,46 +101,108 @@ def block_tables(self, ltable, rtable, l_block_attr, r_block_attr,
         cm._validate_metadata_for_table(ltable, l_key, 'ltable', logger, verbose)
         cm._validate_metadata_for_table(rtable, r_key, 'rtable', logger, verbose)
 
+        # # update the progress bar
+        if show_progress:
+            prog_bar.update() # indicating completion of validation
+   
         # do blocking
 
         # # remove nans: should be modified based on missing data policy
         l_df, r_df = rem_nan(ltable, l_block_attr), rem_nan(rtable, r_block_attr)
 
+        # # update the progress bar to indicate completion of NaN removal
+        if show_progress:
+            prog_bar.update()
+
         # # do projection before merge
-        l_proj_attrs = self.get_proj_attrs(l_key, l_block_attr, l_output_attrs)
+        l_proj_attrs = self.get_attrs_to_project(l_key, l_block_attr, l_output_attrs)
         l_df = l_df[l_proj_attrs]
-        r_proj_attrs = self.get_proj_attrs(r_key, r_block_attr, r_output_attrs)
+        r_proj_attrs = self.get_attrs_to_project(r_key, r_block_attr, r_output_attrs)
         r_df = r_df[r_proj_attrs]
-
-        # # use pandas merge to do equi join
-        candset = pd.merge(l_df, r_df, left_on=l_block_attr,
-                           right_on=r_block_attr,
-                           suffixes=('_ltable', '_rtable'))
-
-        # construct output table
-        retain_cols, final_cols = self.output_columns(l_key, r_key,
-                                                      list(candset.columns),
-                                                      l_output_attrs,
-                                                      r_output_attrs,
-                                                      l_output_prefix,
-                                                      r_output_prefix)
-        candset = candset[retain_cols]
-        candset.columns = final_cols
+       
+        # # update the progress bar to indicate completion of projection of tables
+        if show_progress:
+            prog_bar.update()
+
+        # # determine number of processes to launch parallely
+        n_procs = self.get_num_procs(n_jobs, len(l_df) * len(r_df)) 
+        candset = None
+        if n_procs <= 1:
```

Comment:
n_jobs = -1 => n_procs = n_cpus
This computation is being done in get_num_procs.